### ISSUE TITLE
Allow sending CONNECT frame without `host` header

### ISSUE THREAD

**Issue by Shusek** [joffrey-bion/krossbow#122]
**Problem/use case that this feature is meant to solve/enable**
Currently the server I am connecting to has a little problem with `host` header inside CONNECT frame. If header exist, i got `Bad Connect` error from server. After cut this header everything works fine. 
I am not sure if this header is required by stomp protocol, but some other libraries does not have such a requirement eg. popular library https://github.com/NaikSoftware/StompProtocolAndroid  

**Describe the solution you'd like**
Allow to send connect without host header

**Describe alternatives you've considered**

Is there a way -maybe inconvenient- to achieve your goal at the moment, or a workaround?

At this moment i cut this off this header inside `override suspend fun sendText(frameText: String)`

**Comment by joffrey-bion** [comment]
Hi @Shusek, thanks a lot for the report.

It is very strange that you get such error, because the `host` header is definitely a "MUST" in the STOMP specification:
https://stomp.github.io/stomp-specification-1.2.html#CONNECT_or_STOMP_Frame

In your case, do you have to remove the header completely to make it work, or is it sufficient to set it to the empty string?

One way you can set it to empty string at the moment is to use `customHeaders` in the `connect()` call:

```kotlin
stompClient.connect(url = "...", customHeaders = mapOf(HeaderNames.HOST, ""))
```

There is definitely a missing feature here to customize the host in a better way (users should not need to go through custom headers for that). But I wonder if you actually need no `host` at all.

**Comment by Shusek** [comment]
Unfortunately, I can't send such a header in any form. 
At this moment i'm setting the header to empty string (like in your code example) to avoid regex and then i use inside `KtorWebSocketConnectionAdapter` string cut by:
`wsSession.outgoing.send(Frame.Text(frameText.replace("\nhost:", "")))`
And its works fine, this it's not a big problem for me but look a bit weird.

I rather doubt that it will be fixed inside my backend (old and stable service).

**Comment by joffrey-bion** [comment]
> And its works fine, this it's not a big problem for me but look a bit weird.

I totally understand, it's not ideal. Also it means every single frame going through your websocket will go through this `replace()` call, which is really not nice.

I don't believe this feature can hurt users of STOMP 1.1+ (they just shouldn't set this header to `null`) but on the flip side it can definitely help people in your situation. So I will make it nullable.

**Comment by joffrey-bion** [comment]
Released in 2.1.0. You can now call `stompClient.connect(url, host = null)`.

**Comment by Shusek** [comment]
Everything works fine, thank you very much for help.
